internal/command/jobs/printers.go

```diff
@@ -119,6 +125,79 @@ func printJobDetails(cfg *config.Job, out io.Writer, job *models.Job) error {
 	return nil
 }
 
+func waitForJob(ctx context.Context, cfg *config.JobSubmit, out io.Writer, jobName string) error {
+
+	fmt.Fprintf(out, "Waiting for job execution\n")
+
+	looped := false
+
+	retry := poll.
+		NewPoll().
+		WithDelay(TimeBetweenJobRefresh).
+		WithTimeout(cfg.Timeout)
+
+	lastCursor := ""
+
+	err := retry.Until(func() bool {
+		j, err := getJob(cfg.Job, jobName)
+		if err != nil {
+			fmt.Fprintf(out, "Error getting job: %s", err.Error())
+
+			// We want to keep retrying if the timeout has not been exceeded
+			return false
+		}
+
+		attempt := j.Status.Attempts[0]
+		switch attempt.Phase {
+		case "failed":
+			os.Exit(int(attempt.State.Failed.ExitCode))
```

Comment:
We should update the go-sdk to the latest version.
In that one the `Failed.ExitCode` became a pointer.